<h2>Q5:以參與者的每周學習時間以及特質焦慮量的相關性，探討學習時間是否影響焦慮的生成</h2>

In [9]:
# modules we'll use
import pandas as pd
import numpy as np
import seaborn as sb
import scipy.stats
import statsmodels.stats.weightstats
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression
from scipy.stats import chi2_contingency
# read in our data
med_df = pd.read_csv("C:/Users/ASUS/Desktop/Data Carrard et al. 2022 MedTeach.csv")
med_df.head(5)

id  age  year  sex  glang  part  job  stud_h  health  psyt  jspe  qcae_cog  \
0   2   18     1    1    120     1    0      56       3     0    88        62   
1   4   26     4    1      1     1    0      20       4     0   109        55   
2   9   21     3    2      1     0    0      36       3     0   106        64   
3  10   21     2    2      1     0    1      51       5     0   101        52   
4  13   21     3    1      1     1    0      22       4     0   102        58   

   qcae_aff  amsp  erec_mean  cesd  stai_t  mbi_ex  mbi_cy  mbi_ea  
0        27    17   0.738095    34      61      17      13      20  
1        37    22   0.690476     7      33      14      11      26  
2        39    17   0.690476    25      73      24       7      23  
3        33    18   0.833333    17      48      16      10      21  
4        28    21   0.690476    14      46      22      14      23

<h2>描述統計</h2>

In [4]:
#stud_h stai_t
category = med_df[["stud_h","stai_t"]].copy()
print(category.corr())

          stud_h    stai_t
stud_h  1.000000  0.152052
stai_t  0.152052  1.000000


<h3>
每周學習時間和焦慮量的相關係數為0.152052，
但仍然不能確定兩者是碰巧相關，還是真的相關
</h3>

In [8]:
print(med_df["stud_h"].describe())
print(med_df["stai_t"].describe())


count    886.000000
mean      25.288939
std       15.927875
min        0.000000
25%       12.000000
50%       25.000000
75%       36.000000
max       70.000000
Name: stud_h, dtype: float64
count    886.000000
mean      42.898420
std       11.978458
min       20.000000
25%       34.000000
50%       43.000000
75%       51.000000
max       77.000000
Name: stai_t, dtype: float64


|                |   mean   | std    |  min  | 25%     |  50%     | 75%  | max  |
|  ----          | ----     |  ----   | ---- | ----   |----     |----     |----   |
| Learning time  | 25.2889  | 15.9278 |0   |  12     |  25       |36     | 70  | 
| Anxiety amount |  42.8984 |11.9784 |  20  |34      |43         | 51      |77 |



<h2>檢定統計</h2>

<h3>檢定每周學習時間和焦慮量的關聯性 : 卡方檢定</h3>

H0: 每周學習時間和焦慮量無關

H1: 每周學習時間和焦慮量有關

In [26]:
#alpha = 0.05
chi2_table = pd.DataFrame(columns=['Variable 1', 'Variable 2', 'Chi-Square', 'p-value'])
chi2, p, dof, ex = chi2_contingency(pd.crosstab(med_df["stud_h"], med_df["stai_t"]))
chi2_table = chi2_table.append({'Variable 1': "stud_h", 'Variable 2': "stai_t", 'Chi-Square': chi2, 'p-value': p}, ignore_index=True)
chi2_table

C:\Users\ASUS\AppData\Local\Temp\ipykernel_16168\1146802979.py:4: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  chi2_table = chi2_table.append({'Variable 1': "stud_h", 'Variable 2': "stai_t", 'Chi-Square': chi2, 'p-value': p}, ignore_index=True)


Variable 1 Variable 2   Chi-Square   p-value
0     stud_h     stai_t  3332.034372  0.630795

<h3>
p-value ~= 0.6308 > 0.05 ， 我們不棄卻H0

所以在顯著水準0.05的情況下，每周學習時間和焦慮量無關
</h3>

<h2>延伸分析</h2>
<h3>
既然每周學習時間和焦慮量(心理上的)無關，那麼就會好奇是否和生理健康有關呢

於是，我們可以把每周學習時間和生理健康也做一次卡方檢定
</h3>
H0: 每周學習時間和生理健康無關

H1: 每周學習時間和生理健康有關

In [27]:
chi2_table2 = pd.DataFrame(columns=['Variable 1', 'Variable 2', 'Chi-Square', 'p-value'])
chi2, p, dof, ex = chi2_contingency(pd.crosstab(med_df["stud_h"], med_df["health"]))
chi2_table2 = chi2_table2.append({'Variable 1': "stud_h", 'Variable 2': "health", 'Chi-Square': chi2, 'p-value': p}, ignore_index=True)
chi2_table2

C:\Users\ASUS\AppData\Local\Temp\ipykernel_16168\3357735946.py:3: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  chi2_table2 = chi2_table2.append({'Variable 1': "stud_h", 'Variable 2': "health", 'Chi-Square': chi2, 'p-value': p}, ignore_index=True)


Variable 1 Variable 2  Chi-Square   p-value
0     stud_h     health  293.764489  0.010132

<h3>
p-value ~= 0.0101 < 0.05，我們棄卻H0

所以在顯著水準0.05的情況下，每周學習時間和生理健康有關

</h3>


<h2>
結論
</h2>
<h3>
影響健康的層面可能有很多種，而學習時間的長短對醫學生健康的相關性

和焦慮量無關，而是跟生理健康有關
</h3>
